app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,49 @@
+class FriendshipsController < ApplicationController
+  def new
+    @users = User.all_except(current_user)
+  end
+
+  def create
+    user = current_user
+      @friend = Friendship.create(user_id: user.id, friend_id: params[:friend_id])
+      if @friend.save
+        flash[:success] = "Request sent successfully"
+        redirect_to new_friendship_path
+      else
+        flash[:error] = "Already Friends"
+        redirect_to new_friendship_path
+      end
+  end
+
+  def show
+    @friend = Friendship.find(params[:id])
+  end
+
+  def edit
+    user = current_user
+    @friend = Friendship.find_by_user_id_and_friend_id(params[:id], user.id)
+  end
+  
+  def update
+    @friend = Friendship.find(params[:id])
+      if @friend.update(friend_params)
+        flash[:success] = "Request Accepted"
+        redirect_to friendships_path
+      else
+        flash[:error] = "Something went wrong"
+        render "edit"
+      end
+  end
+
+  def index
+    user = current_user
+    @friends = User.where(id: Friendship.where(user_id: user.id, status: true).pluck(:friend_id))
```

Comment:
this does not contain all friends, I think we discussed this case.

These methods(friends, friend requests etc) should be separated into independent actions in controller.

Reply:
Okay